File: backend/business-service/src/main/java/com/txakurrapp/businessservice/model/Address.java
package com.txakurrapp.businessservice.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String streetName;
    private Integer number;
    private String apt;
    private String otherInfo;
    private String postalCode;
    private String city;
    private String province;
    private String country;

    @OneToOne(mappedBy = "address", orphanRemoval = true)
    @JsonIgnore
    private BusinessOwner businessOwner;

    public Address() {
    }

    public Address(String streetName, Integer number, String apt, String otherInfo, String postalCode, String city, String province, String country) {
        this.streetName = streetName;
        this.number = number;
        this.apt = apt;
        this.otherInfo = otherInfo;
        this.postalCode = postalCode;
        this.city = city;
        this.province = province;
        this.country = country;
    }

//  Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getStreetName() {
        return streetName;
    }

    public void setStreetName(String streetName) {
        this.streetName = streetName;
    }

    public Integer getNumber() {
        return number;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }

    public String getApt() {
        return apt;
    }

    public void setApt(String apt) {
        this.apt = apt;
    }

    public String getOtherInfo() {
        return otherInfo;
    }

    public void setOtherInfo(String otherInfo) {
        this.otherInfo = otherInfo;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }
}

File: frontend/txakurrapp/src/app/classes/petowner/address.ts
export class Address {

    constructor(
        private _id: number,
        private _streetName: string,
        private _number: number,
        private _apt: string,
        private _otherInfo: string,
        private _postalCode: string,
        private _city: string,
        private _province: string,
        private _country: string,
        private _owner: Object
    ){}


    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get streetName(): string {
        return this._streetName;
    }
    public set streetName(value: string) {
        this._streetName = value;
    }
    public get number(): number {
        return this._number;
    }
    public set number(value: number) {
        this._number = value;
    }
    public get apt(): string {
        return this._apt;
    }
    public set apt(value: string) {
        this._apt = value;
    }
    public get otherInfo(): string {
        return this._otherInfo;
    }
    public set otherInfo(value: string) {
        this._otherInfo = value;
    }
    public get postalCode(): string {
        return this._postalCode;
    }
    public set postalCode(value: string) {
        this._postalCode = value;
    }
    public get city(): string {
        return this._city;
    }
    public set city(value: string) {
        this._city = value;
    }
    public get province(): string {
        return this._province;
    }
    public set province(value: string) {
        this._province = value;
    }
    public get country(): string {
        return this._country;
    }
    public set country(value: string) {
        this._country = value;
    }
    public get owner(): Object {
        return this._owner;
    }
    public set owner(value: Object) {
        this._owner = value;
    }

}


